src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -431,11 +432,11 @@ private static void formatUnsignedIntUTF16(int val, int shift, byte[] buf, int l
     public static String toString(int i) {
         int size = DecimalDigits.stringSize(i);
         if (COMPACT_STRINGS) {
-            byte[] buf = new byte[size];
+            byte[] buf = StringConcatHelper.newArray(size);
             DecimalDigits.getCharsLatin1(i, size, buf);
             return new String(buf, LATIN1);
         } else {
-            byte[] buf = new byte[size * 2];
+            byte[] buf = StringConcatHelper.newArray(size << 1);
```

Comment:
@RogerRiggs I recommended StringConcatHelper because that was used in StringLatin1.

For StringUTF16, it has its own newBytesFor:
```suggestion
            byte[] buf = StringUTF16.newBytesFor(size);
```
Or we can factor out a `StringLatin1.newBytesFor` too and encapsulate `newArray` usage.

Reply:
```
    public static byte[] newBytesFor(int len) {
        return new byte[newBytesLength(len)];
    }
```
StringUTF16.newBytesFor uses new byte[], which is not the purpose of this PR